### DISCUSSION TITLE
wire:submit.prevent is not working, page is getting reloaded.

### DISCUSSION THREAD

**Discussion by Ravenna** [livewire/livewire#3466] [1. Help] [answered]
I have a pretty simple livewire component that helps a user store a vanity URL

`<form wire:submit.prevent="save">
	    <div class="form-group form-inline">
	    	<span>{{ config('app.url') }}/request/</span>
	    	<input type="text" id="vanity" wire:model.defer="vanityUrl" name="vanityUrl">
	    </div>
	    <div class="form-group">
	    	<button type="submit" class="btn btn-primary">@if($saved) Saved @else Save @endif</button>
	    	<button type="button" id="copy-vanity" class="btn btn-secondary">Copy</button>
	    </div>
	</form>
`

And the save action is pretty simple as well.

`    public function save()
    {
    	$user = Auth::user();
    	$this->validate();
    	$user->update(['vanity_slug' => Str::slug($this->vanityUrl)]);
    	$this->vanityUrl = Str::slug($this->vanityUrl);
    	$this->saved = true;
    }`

This works locally but not in production.  I am using Forge, php7.4, L8, its all the same between the environments except the URLs.  I have livewire:scripts and livewire:styles loaded, and again it works locally but not in prod.

In prod, the page reloads.  I have been trying to figure this out for a couple days now... driving me crazy.  TIA

**Comment by Qirel** [comment]
Hi.

Sounds like Livewire's scripts isn't loaded in your production environment. 

First step, ensure Livewire is actually installed in your production environment. You can do that by checking composer in production `composer show -D` and look for `livewire/livewire` .  

If Livewire is actually installed, its typically a cache issue - more often than not, this issue is caused when Laravel caches your views (specifically the layout where you put the Livewire scripts) before it has registered the `@livewireScripts`/`@livewireStyles` directives. You can fix this by running `php artisan view:clear`.

**Reply by Ravenna** [reply]
Thanks @Qirel, 

I tried that as well as PHP artisan optimize.  Neither resolved the issue.  

Livewire is installed and linked, the script loads just fine. That is why this is so confounding. I do have Alpine 3 loaded but I that shouldn't be causing this.  

The form element gets a wire:id, wire:initial-data, and the wire:prevent.submit.

**Comment by joshhanley** [comment]
@Ravenna can you confirm Livewire is actually running in production?

Load your production site in your browser and go to devtools > console, then type in `Livewire`. 
You should see the Livewire JS object come up.

If not, that suggests the scripts aren't loading properly.

Hope this helps!

**Reply by Ravenna** [reply]
Livewire is indeed running.

**Comment by Ravenna** [comment, marked as answer]
@Qirel thanks for the help on SO.  The issue is that if you are running Alpine 3+ you have to have Livewire 2.5+.

**Reply by nkmswot** [reply]
How to resolve this?
